### PR TITLE
Cleanup code identified in code review

### PR DESCRIPTION
**Pull request by nbougalis**

### CODE REVIEW

File: src/ripple/app/misc/HashRouter.cpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ HashRouter::shouldRelay (uint256 const& key)
     if (!s.shouldRelay(suppressionMap_.clock().now(), holdTime_))
         return boost::none;
 
-    return std::move(s.peekPeers());
+    return std::move(s.releasePeerSet());
```
Comment (vinniefalco) [inline src/ripple/app/misc/HashRouter.cpp]: I doubt the `std::move` is necessary. @HowardHinnant ?

Comment (nbougalis) [inline src/ripple/app/misc/HashRouter.cpp]: It _probably_ isn't.

Comment (seelabs) [inline src/ripple/app/misc/HashRouter.cpp]: We definitely don't want the `move` here. It'll prevent copy elision.

Comment (nbougalis) [inline src/ripple/app/misc/HashRouter.cpp]: How can you copy-elide? You're constructing a `boost::optional<...>`

Comment (HowardHinnant) [inline src/ripple/app/misc/HashRouter.cpp]: I had missed the return type.  Agreed, it won't prevent copy elision.  The move is not harmful, just unnecessary.

Comment (nbougalis) [inline src/ripple/app/misc/HashRouter.cpp]: The `std::move` has been removed.

**Comment by nbougalis** [comment]
Merged as 44167a6bcba81a095dcff7844b6f25b5bca04708.

**Pull request closed without merging**